### ISSUE TITLE
Improve `\JSONParseArrayValuesMap`

### ISSUE THREAD

**Issue by jasperhabicht** [jasperhabicht/jsonparse#32]
It would be a great improvement to be able to select multiple values at once using `\JSONParseArrayValuesMap` as this could be used to parse data in tabular content. 

It wouls also be nice to be able to nest this function which might be tricky.

The command (as well as `\JSONParseArrayValues` and `\JSONParseArrayCount` ) are a bit sloppy in the selection process which should also be fixes in the course of the development.

**Comment by jasperhabicht** [comment]
It is now possible to select and retrieve multiple values at once. The commands now only select exact values.

It is not yet possible to use this functionality insite `tabular`.

It is not possible to nest `\JSONParseArrayValuesMap` inside each other.

**Comment by jasperhabicht** [comment]
It is now possible to use this functionality with `tabular`.

**Comment by jasperhabicht** [comment]
Maybe, instead of nesting, the package could just provide some kind of for loop that maps over all items in an array returning just the index. But this would require to get the last index before starting the loop.

**Comment by Zumiha** [comment]
Also question, is there issue with pdfLaTeX compiler regarding \JSONParseArrayValuesMap?
During testing it refuses to output anything, but there is no compile error and other stuff like \JSONParseValue for the json works perfectly fine.

**Comment by jasperhabicht** [comment]
> Also question, is there issue with pdfLaTeX compiler regarding \JSONParseArrayValuesMap?
> 
> During testing it refuses to output anything, but there is no compile error and other stuff like \JSONParseValue for the json works perfectly fine.

Could you try to come up with a code example of your issue? Also, please consider filing a separate issue. This way, it is easier for me to help you. Thank you!

**Comment by jasperhabicht** [comment]
With the last update (0.9.11), an alternative command to parse arrays item-by-item is implemented that internally uses `\int_step:nnn` and accepts code of an inline function. The following example illustrates its use:

```tex
\documentclass{article}
\usepackage{jsonparse}

\begin{filecontents}[noheader]{data.json}
{
  "items": [
    {
      "id": "abc",
      "value": 1
    },
    {
      "id": "def",
      "value": 2
    },
  ]
}
\end{filecontents}

\JSONParseFromFile{\jsondata}{data.json}

\begin{document}
    \begin{tabular}{ c c }
        \JSONParseArrayMapInline{\jsondata}{items}{%
            \JSONParseValue{\jsondata}{items[#1].id} & 
            \JSONParseValue{\jsondata}{items[#1].value} \\
        }
    \end{tabular}
\end{document}
```

The output is:

![image](https://github.com/user-attachments/assets/801ad5cf-3b48-4cf7-b4ff-88706b32fa75)

**Comment by jasperhabicht** [comment]
Currently, it seems difficult to access the names of keys when looping through arrays. Consider the following JSON data:

```json
{ "matrix": [
    { "key_a" : "value_a" },
    { "key_b" : "value_b" },
    { "key_c" : "value_c" }
] }
```

This can't easily be converted into a table that contains the keys in one column and the values in the other because it is difficult to access the keys:

| key | value |
| --- | --- |
| key_a | value_a |
| key_b | value_b |
| key_c | value_c |

**Comment by jasperhabicht** [comment]
Probably a redesign of `\JSONParseKeys` could be reasonable. Currently, this command generates a JSON array and store it in a token variable. While this may be needed in certain circumstances, it is not very useful for further processing. 

It would be much better to have a (new?) command that stores the keys as property list or sequence in order to be able to easily access items in loops.

**Comment by jasperhabicht** [comment]
New commands available as of version 0.9.12.
